lib/manageiq/appliance_console/message_configuration.rb

```diff
@@ -0,0 +1,134 @@
+require 'active_support/core_ext/module/delegation'
+require 'pathname'
+
+module ManageIQ
+  module ApplianceConsole
+    class MessageConfiguration
+      attr_reader :username, :password,
+                  :miq_config_dir_path, :config_dir_path, :sample_config_dir_path,
+                  :tools_log4_properties_path, :tools_log4_properties_sample_path,
+                  :client_properties_path,
+                  :keystore_dir_path, :truststore_path, :keystore_path,
+                  :messaging_yaml_sample_path, :messaging_yaml_path
+
+      BASE_DIR                          = "/opt/kafka".freeze
+      LOGS_DIR                          = "#{BASE_DIR}/logs".freeze
+      CONFIG_DIR                        = "#{BASE_DIR}/config".freeze
+      SAMPLE_CONFIG_DIR                 = "#{BASE_DIR}/config-sample".freeze
+      MIQ_CONFIG_DIR                    = "/var/www/miq/vmdb/config".freeze
+
+      def initialize(options = {})
+        @username, @password = nil
+
+        @username = options[:username] || "admin"
+        @password = options[:password]
+
+        @miq_config_dir_path               = Pathname.new(MIQ_CONFIG_DIR)
+        @config_dir_path                   = Pathname.new(CONFIG_DIR)
+        @sample_config_dir_path            = Pathname.new(SAMPLE_CONFIG_DIR)
+
+        @tools_log4_properties_path        = @config_dir_path.join("", "tools-log4j.properties")
+        @tools_log4_properties_sample_path = @sample_config_dir_path.join("", "tools-log4j.properties")
+        @client_properties_path            = @config_dir_path.join("", "client.properties")
+
+        @keystore_dir_path                 = @config_dir_path.join("", "keystore")
+        @truststore_path                   = @keystore_dir_path.join("", "truststore.jks")
+        @keystore_path                     = @keystore_dir_path.join("", "keystore.jks")
+
+        @messaging_yaml_sample_path        = @miq_config_dir_path.join("", "messaging.kafka.yml")
+        @messaging_yaml_path               = @miq_config_dir_path.join("", "messaging.yml")
+      end
+
+      def create_tools_log_config
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        FileUtils.cp(@tools_log4_properties_sample_path, @tools_log4_properties_path) unless file_found?(@tools_log4_properties_path)
```

Comment:
You shouldn't need the `@` here since it's an `attr_reader` (same throughout the PR).

Reply:
What do you think of `tools_log4_properties_path.exist?` (Pathname#exist?) rather than a new `file_found?` method?

Reply:
file_found? also logs a message (not saying it should, just that's why there's a separate method)

Reply:
Correct, @Fryguy the _file_found?_ method also logs the message letting  the user know the file already exists.
@bdunne I will remove the unnecessary `@`. I just like to leave them so when reading the code the scope of the variable is immediately  evident. But if it is more acceptable style I',m  OK with removing them.

Updated by commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/130/commits/a1a2b4a5cf7ab94cfd249c654ae45ce324bdc34f